playground/update.sh

```diff
@@ -20,18 +20,15 @@ gopherjs build -m
 # The GOPATH workspace where the GopherJS project is.
 gopherjsgopath=$(go list -f '{{.Root}}' github.com/gopherjs/gopherjs)
 
-rm -r pkg/
+rm -rf pkg/
 
 # Use an empty GOPATH workspace with just gopherjs,
 # so that all the standard library packages get written to GOROOT/pkg.
 export GOPATH="$tmp/gopath"
 mkdir -p "$GOPATH"/src/github.com/gopherjs/gopherjs
 cp -a "$gopherjsgopath"/src/github.com/gopherjs/gopherjs/* "$GOPATH"/src/github.com/gopherjs/gopherjs
 
-gopherjs install -m github.com/gopherjs/gopherjs/js github.com/gopherjs/gopherjs/nosync
-mkdir -p pkg/github.com/gopherjs/gopherjs
-cp "$GOPATH"/pkg/*_js_min/github.com/gopherjs/gopherjs/js.a pkg/github.com/gopherjs/gopherjs/js.a
-cp "$GOPATH"/pkg/*_js_min/github.com/gopherjs/gopherjs/nosync.a pkg/github.com/gopherjs/gopherjs/nosync.a
+mkdir pkg
```

Comment:
Yes, worked fine locally for me. But I suspect [my answer](https://github.com/gopherjs/gopherjs.github.io/issues/69#issuecomment-384009913) has something to do with that?

Reply:
I can't reproduce your results.

If I remove the `js.a.js` file from pkg and use `gopherjs serve`, the playground fails to run the initial program.

```
playground $ rm ./pkg/github.com/gopherjs/gopherjs/js.a.js 
playground $ gopherjs serve
serving at http://localhost:8080 and on port 8080 of any available addresses
```

![image](https://user-images.githubusercontent.com/1924134/39211640-43051ad2-47da-11e8-80e0-115ac6b4d689.png)

Maybe you didn't clear your browser's cache?

Reply:
I did the following (recreating the root-owned `$GOROOT` for fair comparison):

```
cd $GOROOT
sudo chown -R root:root .
cd `mktemp -d`
export GOPATH=$PWD
export PATH=$GOPATH/bin:$PATH
go get github.com/gopherjs/gopherjs
go get github.com/gopherjs/gopherjs.github.io/playground
cd $(go list -f "{{.Dir}}" github.com/gopherjs/gopherjs.github.io/playground)
git fetch origin pull/70/head
git checkout -q FETCH_HEAD
which gopherjs
go generate
gopherjs serve
```

Then went to http://localhost:8080/github.com/gopherjs/gopherjs.github.io/playground/ and did an "Empty Cache and Hard Reload" in Chrome and that page, which imports `js`, works fine for me: I see the alert as well as the console log and the in-page "console" log.

Am I following the right steps?

For me `github.com/gopherjs/gopherjs/js` is compiled into `playground.js`:

```
$ grep '^\$packages\["github.com/gopherjs/gopherjs/js"\]' playground.js | cut -c 1-80
$packages["github.com/gopherjs/gopherjs/js"]=(function(){var $pkg={},$init,A,B,I
```

Reply:
Thanks for providing more information.

Can you check, after you do `go generate`, are there 2 files in [./pkg/github.com/gopherjs/gopherjs](https://github.com/gopherjs/gopherjs.github.io/tree/cbdec896dfc8a335278926fc7d6a7e2c43ccdb05/playground/pkg/github.com/gopherjs/gopherjs) directory?

Based on the steps you're following... Maybe what's happening is that you _already_ have a built `js.a` and `nosync.a` in your GOROOT/pkg from before (when your GOROOT was user-writeable), and it gets copied from there?

Reply:
> For me github.com/gopherjs/gopherjs/js is compiled into playground.js:

That's expected, it's there for the playground itself. It doesn't mean it would be used from there during compilation of playground programs.

Reply:
> That's expected, it's there for the playground itself. github.com/gopherjs/gopherjs/compiler will also be there, etc. But it doesn't mean it would be used from there during compilation of playground programs.

Good point.

> Can you check, after you do go generate, are there 2 files in ./pkg/github.com/gopherjs/gopherjs directory?

Yes there are.

I've also just run the steps above but with a writable `GOROOT`; success again.

But I don't think this is surprising because after the `gopherjs install` in the fake `GOROOT` we do:

```
cp -a "$GOROOT"/pkg/*_js_min/* pkg/
cp -a "$GOROOT"/pkg/*_amd64_js_min/* pkg/
```

and because [some of the natives](https://github.com/gopherjs/gopherjs/blob/8dffc02ea1cb8398bb73f30424697c60fcf8d4c5/compiler/natives/src/math/math.go#L6) depend on `github.com/gopherjs/gopherjs/js` it also gets installed (because `gopherjs` installs all dependencies, unlike Go 1.10).

I think this adds up now?

Incidentally, do the steps in https://github.com/gopherjs/gopherjs.github.io/pull/70#discussion_r183882766 not work for you?